### PR TITLE
Reveal map at start

### PR DESCRIPTION
**Pull request by Ragnarok77-factorio**
### Brief description of the changes:
Reveal map at start for 2k distance, first both team, then 3 minutes later, do a second reveal so south has the map too as the mirror was not yet done, so to see ores in south you need this second scan.
Toggleable by admin

Vote :
https://discord.com/channels/823696400797138974/823771211421974579/1127593258940366859
image
+ second one : 
https://discord.com/channels/823696400797138974/823771211421974579/1113151225098162358
image

### Tested Changes:
- [x] I've tested the changes locally
- [ ] I've not tested the changes.

### CODE REVIEW

File: maps/biter_battles_v2/init.lua [hunk 1]
```diff
@@ -176,6 +180,38 @@ function Public.draw_structures()
...
+local decrement_timer_token = Token.get_counter() + 1 -- predict what the token will look like
+decrement_timer_token = Token.register(
+    function()
+        local reveal_time_left = global.reveal_time_left - 1
+        if reveal_time_left > 0 then
+            Task.set_timeout_in_ticks(180, decrement_timer_token)
+            global.reveal_time_left = reveal_time_left
+        end
+    end
+)
```
Comment (TheBigZet) [inline maps/biter_battles_v2/init.lua:201]: What is it for?

Comment (Ragnarok77-factorio) [reply maps/biter_battles_v2/init.lua:201]: Thanks for review ! it is because on bb, when you reveal the map at start, only the ores of north are revealed, so the whole map is revealed, but south has no ores shown : 
image

Because I guess the mirroring comes later, so as a workaround, we reveal the whole map, then 3 minutes later (which is how long we need to wait, I reveal again), and then all the ores of south are revealed . 
I didn't want to use tick because it's very bad perf wise, and I didn't find any better solution, so I used what you did for reroll.

Comment (TheBigZet) [reply maps/biter_battles_v2/init.lua:201]: Let me rephrase it. Do you use `global.reveal_time_left` anywhere else?

Comment (Ragnarok77-factorio) [reply maps/biter_battles_v2/init.lua]: Thanks @TheBigZet , you are correct, it was useless, same for decrement and so on, it's not even called, and I only need to execute a task in 3 minutes, no need to check every second or use a decrement function, as I don't use it to show a time in gui like in reroll PR.. 

I think it's resolved..? Feel free to open it again if I forgot something !

File: maps/biter_battles_v2/init.lua [hunk 2]
```diff
@@ -179,6 +183,27 @@ function Public.draw_structures()
...
+
+function Public.reveal_map()
+	if global.bb_settings["bb_map_reveal_toggle"] then
+		local surface = game.surfaces[global.bb_surface_name]
+		local radius = 2000
+		game.forces["north"].chart(surface, {{0-radius, 500}, {0+radius, -500}}) 
+		game.forces["south"].chart(surface, {{0-radius, 500}, {0+radius, -500}}) 
+		Task.set_timeout_in_ticks(10800, reveal_token)
```
Comment (developer-8) [inline maps/biter_battles_v2/init.lua:203]:
```suggestion
		local width = 2000 -- for one side
		local height = 500 -- for one side
		for x = 1, width, 32 do
			for y = 33, height, 32 do
				game.forces["spectator"].chart(surface, {{-x, -y}, {x, -y+32}})
				game.forces["spectator"].chart(surface, {{-x, y-32}, {x, y}})
			end
		end
```
this way no need to rechart it again after

Comment (developer-8) [inline maps/biter_battles_v2/init.lua:203]:
```suggestion
		local width = 2000 -- for one side
		local height = 500 -- for one side
		for x = 16, width, 32 do
			for y = 16, height, 32 do
				game.forces["spectator"].chart(surface, {{-x, -y}, {-x, -y}})
				game.forces["spectator"].chart(surface, {{x, -y}, {x, -y}})
				game.forces["spectator"].chart(surface, {{-x, y}, {-x, y}})
				game.forces["spectator"].chart(surface, {{x, y}, {x, y}})
			end
		end
```
in this way rechart is not required

Comment (Ragnarok77-factorio) [reply maps/biter_battles_v2/init.lua:203]: Wow, what a great update !  Thanks a lot @developer-8  I was wondering if we could have avoid this double rechart... You found it.

**Comment by Ragnarok77-factorio** [comment]
Code updated, test OK, much cleaner now.

File: maps/biter_battles_v2/init.lua [hunk 3]
```diff
@@ -124,6 +124,7 @@ function Public.initial_setup()
 		["only_admins_vote"] = false,		--Are only admins able to vote on the global difficulty?
 		--MAP SETTINGS--
 		["new_year_island"] = false,
+		["bb_map_reveal_toggle"] = true,
```
Comment (developer-8) [inline maps/biter_battles_v2/init.lua:127]:
```suggestion
		["bb_map_reveal_toggle"] = false,
```

Comment (developer-8) [reply maps/biter_battles_v2/init.lua:127]: I don't like that it's on by default, can it turned off? like reroll. So that when testing it doesn’t generate the entire map every time

Comment (Ragnarok77-factorio) [reply maps/biter_battles_v2/init.lua:127]: I mean, can't you just edit locally before starting the scenario to false..? Making it false by default is...

I mean vote was to get this reveal, so every time we restart server, admins need to think of enabling it.. :s

Comment (developer-8) [reply maps/biter_battles_v2/init.lua:127]: Yes, I'm counting on the fact that one day the admins will forget and won't remember about it anymore

Comment (Ragnarok77-factorio) [reply maps/biter_battles_v2/init.lua:127]: I mean , some players were saying, area too big, don't like the reveal, and so on, but no poll was made to change that. will make a message in #general to see if it was just isolated cases or is there an issue with it.

**Comment by Ragnarok77-factorio** [comment]
@XVhc6A  or @developer-8 or @cogito123 or another dev, could you please review the PR and tell if there is any issue/change to do or if it's fine ? 

I need another dev approval before merging.

**Review by XVhc6A — APPROVED** [review]
changes look good to me, even though i disagree with 2k map reveal : p

this has been tested thoroughly on the server.
